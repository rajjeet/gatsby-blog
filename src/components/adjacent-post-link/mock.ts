export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
