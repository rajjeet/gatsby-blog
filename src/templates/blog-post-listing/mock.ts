import { mockPosts } from '../../fixtures/testing';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  data: {
    posts: {
      edges: mockPosts,
    },
  },
  pageContext: {
    numOfPages: 2,
    currentPage: 3,
    tag: 'Sample Tag',
  },
});
