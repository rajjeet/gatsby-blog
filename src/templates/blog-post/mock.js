import { createMockPosts } from '../../utils/testing';

export const makeProps = () => ({
  className: {},
  data: {
    post: createMockPosts[1].node,
  },
});
