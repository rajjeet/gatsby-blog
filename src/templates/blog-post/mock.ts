import { TProps } from './types';
import { mockPost } from '../../fixtures/testing';

export const makeProps = ({ dateModified = null } = {}): TProps => {
  if (dateModified) {
    mockPost.frontmatter.dateModified = dateModified;
  }
  return {
    data: {
      post: { ...mockPost },
    },
  };
};
