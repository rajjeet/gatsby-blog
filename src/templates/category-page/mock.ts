import { createMockPosts } from '../../utils/testing';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  data: {
    posts: {
      edges: createMockPosts,
    },
  },
  pageContext: {
    category: 'Self Improvement',
    numOfPages: 2,
    currentPage: 3,
    paginationSlug: '/self-improvement/',
  },
});
