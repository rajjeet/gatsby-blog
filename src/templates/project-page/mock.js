import { createMockPosts, createMockProjects } from '../../utils/testing';

export const makeProps = () => ({
  data: {
    project: createMockProjects[0].node,
    posts: {
      edges: createMockPosts,
    },
  },
});
