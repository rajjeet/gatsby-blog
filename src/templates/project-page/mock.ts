import { TProps } from './types';
import { mockPosts } from '../../fixtures/posts';
import { mdxBody } from '../../fixtures/mdx';

export const makeProps = (): TProps => ({
  data: {
    project: {
      frontmatter: {
        title: 'Condo Landing Pages',
        description: 'An automated process for building fast, responsive landing pages for pre-construction condos.',
        techStackTags: [
          {
            type: 'AWS',
            label: 'AWS CloudFormation',
          },
        ],
        links: [
          {
            label: 'Github Code',
            value: 'https://github.com/rajjeet/charismacondos',
          },
        ],
      },
      body: mdxBody,
    },
    posts: {
      edges: mockPosts,
    },
  },
});
