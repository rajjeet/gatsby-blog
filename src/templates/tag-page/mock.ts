import { createMockPosts } from '../../utils/testing';
import { TProps } from './index';

export const makeProps = (): TProps => ({
  data: {
    posts: {
      edges: createMockPosts,
    },
  },
  pageContext: {
    limit: 5,
    skip: 2,
    numOfPages: 2,
    numOfPosts: 10,
    currentPage: 3,
    tag: 'Life Skills',
    paginationSlug: '',
  },
});
