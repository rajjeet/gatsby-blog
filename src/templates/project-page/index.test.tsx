import React from 'react';
import { render } from '@testing-library/react';
import ProjectPage from './index';
import { makeProps } from './mock';

describe('ProjectPage', () => {
  it('should have the project title', () => {
    const { getByText } = render(<ProjectPage {...makeProps()} />);
    expect(getByText('Condo Landing Pages')).toBeDefined();
  });

  it('should have the project description', () => {
    const { getByText } = render(<ProjectPage {...makeProps()} />);
    expect(getByText(/An automated process for building fast/)).toBeDefined();
  });

  it('should have the tech stack tag listings', () => {
    const { getByText } = render(<ProjectPage {...makeProps()} />);
    expect(getByText('AWS CloudFormation')).toBeDefined();
  });

  it('should have a simple link listing section', () => {
    const { getByText } = render(<ProjectPage {...makeProps()} />);
    expect(getByText('Github Code')).toBeDefined();
    // expect(getByText('Links')).toBeDefined();
  });

  it('should have a simple post listing section', () => {
    const { getByText } = render(<ProjectPage {...makeProps()} />);
    expect(getByText('Adding a Project Section to My Website')).toBeDefined();
    expect(getByText('Blog Posts')).toBeDefined();
    expect(getByText('State Management using React Hooks')).toBeDefined();
  });
});
