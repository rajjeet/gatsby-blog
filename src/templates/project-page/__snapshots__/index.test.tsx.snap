// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPage should render 1`] = `
<DocumentFragment>
  <div
    class="sc-htpNat faJHDz"
  >
    <div
      class="sc-bwzfXH bmdiOC"
    >
      <a
        aria-label="See Home"
        href="/"
      >
        <div
          class=" gatsby-image-wrapper"
          style="position: relative; overflow: hidden; width: 3em;"
        >
          <div
            style="width: 100%; padding-bottom: 100%;"
          />
          <picture>
            <source
              sizes=""
              srcset=""
            />
            <img
              alt="Ortmesh logo"
              loading="lazy"
              sizes=""
              src="https://via.placeholder.com/250x180"
              srcset=""
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="https://via.placeholder.com/250x180" alt="Ortmesh logo" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
      </a>
      <a
        href="/"
      >
        <h2>
          Ortmesh
        </h2>
        <p>
          Write code that matters
        </p>
      </a>
      <button
        class="sc-bdVaJa eSxrIN"
      >
        About
      </button>
    </div>
    <div
      class="sc-gzVnrw jgilSY"
    >
      <h1>
        Condo Landing Pages
      </h1>
      <div>
        An automated process for building fast, responsive landing pages for pre-construction condos.
      </div>
      <div
        class="sc-bZQynM kZGmCz"
      >
        <span>
          Tech Tags:
        </span>
        <div>
          AWS CloudFormation
        </div>
      </div>
      <div
        class="sc-htoDjs LtFSS"
      >
        <div>
          <div>
            <h3>
              Links
            </h3>
            <ul
              class="sc-ifAKCX eaAsdb"
            >
              <a
                aria-label="Github Code"
                href="https://github.com/rajjeet/charismacondos"
                rel="noopener noreferrer"
                target="_blank"
              >
                <li>
                  Github Code
                </li>
              </a>
            </ul>
          </div>
          <div>
            <h3>
              Blog Posts
            </h3>
            <ul
              class="sc-ifAKCX eaAsdb"
            >
              <a
                aria-label="See post: Adding a Project Section to My Website"
                class="sc-bxivhb iVfRmq"
                href="/adding-project-section-to-website/"
              >
                <li>
                  <h4>
                    Adding a Project Section to My Website
                  </h4>
                  <div>
                    <span
                      class="sc-EHOje kjAiMu"
                    >
                      11 May, 2019
                    </span>
                    <span>
                      2 mins read
                    </span>
                  </div>
                </li>
              </a>
              <a
                aria-label="See post: State Management using React Hooks"
                class="sc-bxivhb iVfRmq"
                href="/state-management-react-hooks/"
              >
                <li>
                  <h4>
                    State Management using React Hooks
                  </h4>
                  <div>
                    <span
                      class="sc-EHOje kjAiMu"
                    >
                      02 May, 2019
                    </span>
                    <span>
                      4 mins read
                    </span>
                  </div>
                </li>
              </a>
              <a
                aria-label="See post: Build A Super Customized and Performant Blog"
                class="sc-bxivhb iVfRmq"
                href="/build-super-customized-performant-blog/"
              >
                <li>
                  <h4>
                    Build A Super Customized and Performant Blog
                  </h4>
                  <div>
                    <span
                      class="sc-EHOje kjAiMu"
                    >
                      21 February, 2019
                    </span>
                    <span>
                      3 mins read
                    </span>
                  </div>
                </li>
              </a>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
